### PR TITLE
Overlay & Styled-components 4.4.0

### PR DESCRIPTION
**Pull request by luckylwk**
✅ **DONE**:
- [x] Overlay component for Full page overlay and in-side parent overlay
- [x] Styled Components 4.4.0

Because of issues that nextjs has in development (not in production!) with the server-side-rendering. See: 
- [release notes](https://github.com/styled-components/styled-components/releases/tag/v4.4.0)
- [bug](https://github.com/styled-components/styled-components/pull/2701)

image

⏹ **TODO**:
- [ ] Test in NextJS setup
- [ ] Test in App
- [ ] Bump to `v1.0.1`

### CODE REVIEW

**Comment by luckylwk** [comment]
@jonashaefele can you have a look at my `Overlay` component? I am still getting my head around the hooks (`useContext`) but I like how it works and how we have the `Expandable` ability. Not sure if I used it correctly in the Readme for the `Overlay`

File: src/components/Overlay/Overlay.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+Overlay.defaultProps = {
+  position: 'fixed',
+  opacity: '0.97',
+};
+
+Overlay.displayName = 'Overlay';
+
+// ---------------------------
+
+const NavOverlay = styled(Overlay)`
```
[the remark this answers is not in the document]
Comment (luckylwk) [reply src/components/Overlay/Overlay.js]: I'll update that

File: src/components/Overlay/Readme.mdx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import { Flex } from '../Flex';
+import { Box } from '../Box';
+import { Button } from '../Button';
+import { 
+  Expandable, 
+  ExpandableToggle, 
+  ExpandableBody,
+  // ExpandableContext 
+} from '../Expandable';
+import { HeartLoader } from 'components/Loader';
```
Comment (jonashaefele) [inline src/components/Overlay/Readme.mdx]: for docz 2.2:
`import { HeartLoader } from '../Loader';`

Comment (jonashaefele) [reply src/components/Overlay/Readme.mdx]: think that's the only realy change you need to make it work with master/docz2.2

then just run `yarn run build && yarn run prettier:all` again and the rest of the conflicts should be resolved.

**Pull request closed without merging**